android/quest/src/main/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireViewModel.kt

```diff
@@ -609,9 +616,34 @@ constructor(
       val basicResource = defaultRepository.loadResource(resourceId) as Basic?
       bundle.addEntry(Bundle.BundleEntryComponent().setResource(basicResource))
     }
-    questionnaire.cqfLibraryUrls().forEach { library ->
+    questionnaire.cqfLibraryUrls().forEach { url ->
       if (subject.resourceType == ResourceType.Patient) {
-        fhirOperator.evaluateLibrary(library, subject.asReference().reference, null, setOf())
+        val library =
+          defaultRepository.fhirEngine
+            .search<Library> { filter(Library.URL, { value = url }) }
+            .first()
+            .resource
+
+        // Use method to call without expressions
+        val expressionsParam =
+          library.parameter
+            .filter { it.type == Enumerations.FHIRAllTypes.PARAMETERDEFINITION.toCode() }
+            .map { it.name }
+            .toSet()
+
+        // TODO move to Sync
```

Comment:
same above, note on what next steps are here and link to issue if any

Reply:
also, unclear if this is duplicate code, refactor if so

Reply:
Same as above